service/tablet_allocator.cc

```diff
@@ -169,6 +169,7 @@ struct colocated_tablets {
 // to same destination.
 struct migration_tablet_set {
```

Comment:
How about introducing an explicit constructor for this type? I'd rather avoid a situation where we have a `migration_tablet_set` where we forgot to set `group_size` explicitly (either now, or if somebody in the future introduces this kind of a mistake).

Just to be clear, I want to avoid this being legal:

```c++
migration_tablet_set foo { global_tablet_id(...) };
```

Reply:
I want to promote use of designated initializers.

Unfortunately, there is no way to prohibit traditional initializers for those structs, but having restricted constructors prevents using designated initializers.

Reply:
Designated initializers are indeed nice, but they allow you to omit some members (omitted members use the default initializer or empty list initializer, [cppreference link](https://en.cppreference.com/w/cpp/language/aggregate_initialization#Designated_initializers)). If you can define a sensible default value for each field then it's fine, but I'm not sure this is the case here - specifically, for the `group_size` field.